optuna/visualization/_optimization_history.py

```diff
@@ -74,187 +174,75 @@ def objective(trial):
     """
 
     _imports.check()
-
-    if isinstance(study, Study):
-        studies = [study]
-    else:
-        studies = list(study)
-
-    _check_plot_args(studies, target, target_name)
-    return _get_optimization_history_plot(studies, target, target_name, error_bar)
-
-
-def _get_optimization_history_plot(
-    studies: List[Study],
-    target: Optional[Callable[[FrozenTrial], float]],
-    target_name: str,
-    error_bar: bool,
-) -> "go.Figure":
+    _check_plot_args(study, target, target_name)
 
     layout = go.Layout(
         title="Optimization History Plot",
         xaxis={"title": "Trial"},
         yaxis={"title": target_name},
     )
```

Comment:
Should be moved in `_get_optimization_history_plot`.

Reply:
Fixed.